moveit_core/robot_trajectory/include/moveit/robot_trajectory/robot_trajectory.h

```diff
@@ -183,50 +209,86 @@ class RobotTrajectory
    * \param state - current robot state
    * \param dt - duration from previous
    */
-  RobotTrajectory& addSuffixWayPoint(const moveit::core::RobotState& state, double dt)
+  RobotTrajectory& addSuffixWayPoint(const moveit::core::RobotState& state, rclcpp::Duration dt)
   {
     return addSuffixWayPoint(std::make_shared<moveit::core::RobotState>(state), dt);
   }
 
+  [[deprecated("Use of double for duration is deprecated, please pass in rclcpp::Duration")]] RobotTrajectory&
+  addSuffixWayPoint(const moveit::core::RobotState& state, double dt)
+  {
+    return addSuffixWayPoint(state, rclcpp::Duration::from_seconds(dt));
+  }
+
   /**
    * \brief Add a point to the trajectory
    * \param state - current robot state
    * \param dt - duration from previous
    */
-  RobotTrajectory& addSuffixWayPoint(const moveit::core::RobotStatePtr& state, double dt)
+  RobotTrajectory& addSuffixWayPoint(const moveit::core::RobotStatePtr& state, rclcpp::Duration dt)
```

Comment:
Is `rclcpp::Duration` the same size as `double`? We ought to be sure about that before we continue to pass it around by value.

Reply:
Internally it is an int64_t representing nanoseconds. I think that's the same size as `double`.